### PR TITLE
Sub for MatPolyRingZq

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

This PR implements subtraction for MatPolyRingZq.

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/mat_polynomial_ring_zq/arithmetic/sub.rs [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        let poly_ring_mat2 = MatPolynomialRingZq::from((&poly_mat, &modulus2));
+        let poly_ring_mat3 = MatPolynomialRingZq::from((&poly_mat, &modulus3));
+
+        assert!(poly_ring_mat1.sub_safe(&poly_ring_mat2).is_err());
+        assert!(poly_ring_mat3.sub_safe(&poly_ring_mat2).is_err());
+    }
+
+    /// Tests the doc test (availability).
+    #[test]
+    fn doc_test() {
```
Comment (MoogSven) [inline src/integer_mod_q/mat_polynomial_ring_zq/arithmetic/sub.rs:206]: Consider renaming this test to availability. This way it is not a problem if this test and the doc test are not the same.

**Review by MoogSven — APPROVED** [review]
Looks good, just three small remarks
